### PR TITLE
Add support for from_utc_timestamp in Hive views translation

### PR DESCRIPTION
**Pull request by losipiuk**
Requires a Coral release including https://github.com/linkedin/coral/pull/109

### CODE REVIEW

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java [hunk 1]
```diff
@@ -172,4 +172,88 @@ public void testLateralViewJsonTupleAs()
...
+                "  CAST(123 AS tinyint) source_tinyint, " +
+                "  CAST(10123 AS smallint) source_smallint, " +
+                "  CAST(2592000123 AS integer) source_integer, " +
+                "  CAST(2592000123 AS bigint) source_bigint, " +
+                "  CAST(2592000.0 AS float) source_float, " +
+                "  CAST(2592000.123 AS double) source_double, " +
+                "  CAST(2592000.123 AS decimal(10,3)) source_decimal_three," +
+                "  CAST(2592000 AS DECIMAL(10,0)) source_decimal_zero," +
+                "  timestamp '1970-01-30 16:00:00' source_timestamp, " +
+                "  date '1970-01-30 16:00:00' source_date ");
```
Comment (hashhar) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java:190]: Hive allows this? Does it truncate parts smaller than a day?

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java [hunk 2]
```diff
@@ -172,4 +172,88 @@ public void testLateralViewJsonTupleAs()
...
+                "  CAST(ts_bigint as VARCHAR), " +
+                "  CAST(ts_float as VARCHAR), " +
+                "  CAST(ts_double as VARCHAR), " +
+                "  CAST(ts_decimal_three as VARCHAR), " +
+                "  CAST(ts_decimal_zero as VARCHAR), " +
+                "  CAST(ts_timestamp as VARCHAR), " +
+                "  CAST(ts_date as VARCHAR) " +
+                "FROM test_from_utc_timestamp_view"))
+                .containsOnly(row(
+                        "1969-12-31 16:00:00.123",
```
Comment (hashhar) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java:223]: Would be nice to also verify behaviour for very old dates (pre-calendard switch and some negative epoch values).

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java [hunk 3]
```diff
@@ -172,4 +172,88 @@ public void testLateralViewJsonTupleAs()
...
+                        "1970-01-30 16:00:00.123",
+                        "1970-01-30 16:00:00.000",
+                        "1970-01-30 16:00:00.123",
+                        "1970-01-30 16:00:00.123",
+                        "1970-01-30 16:00:00.000",
+                        "1970-01-30 08:00:00.000",
+                        "1970-01-29 16:00:00.000"));
+
+        // check result on Hive
+        assertThat(onHive().executeQuery("SELECT " +
```
Comment (hashhar) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java:235]: There's a useful `assertViewQuery` which may reduce some repetition here (by moving the CAST to STRING into the view definition itself).

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java [hunk 4]
```diff
@@ -172,4 +172,88 @@ public void testLateralViewJsonTupleAs()
...
+                "  CAST(ts_bigint as STRING), " +
+                "  CAST(ts_float as STRING), " +
+                "  CAST(ts_double as STRING), " +
+                "  CAST(ts_decimal_three as STRING), " +
+                "  CAST(ts_decimal_zero as STRING), " +
+                "  CAST(ts_timestamp as STRING), " +
+                "  CAST(ts_date as STRING) " +
+                "FROM test_from_utc_timestamp_view"))
+                .containsOnly(row(
+                        "1969-12-31 16:00:00.123",
```
Comment (hashhar) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java:248]: Extract expected values to a variable - it is errorprone to repeat.

**Review by hashhar — COMMENTED** [review]
Some comments and questions.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java (hunk 1 above)
Comment (losipiuk) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java:190]: Oh - did not notice that. I meant `date '1970-01-30'`. But apparently Hive is not very picky about the literal :)

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java (hunk 2 above)
Comment (losipiuk) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java:223]: I can try. But I am sure this is not an exact copy of Hive's behavior in some corner cases. Just a best-effort translation.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java (hunk 3 above)
Comment (losipiuk) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java:235]: I tried to use that but it does not help as the string values generated by Trino and Hive are different for timestamps (Hive truncates zeros).

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java (hunk 4 above)
Comment (losipiuk) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java:248]: Unfortunatelly they are not exactly the same for Hive and Trino. Formatting of timestamp is slightly different.

**Comment by losipiuk** [comment]
AC

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java (hunk 1 above)
Comment (hashhar) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java]: But it seems we are handling it correctly. Please keep it and add another case with the "correct" literal.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java (hunk 2 above)
Comment (hashhar) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java:212]: Tests to document the behaviour either way would help. But I don't feel too strongly about this.

**Review by hashhar — COMMENTED** [review]
Looks good to me once new Coral version is out.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java (hunk 1 above)
Comment (findepi) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java]: I hope it doesn't matter, _how_ a date value is created. I.e. i do hope `date '1970-01-30 16:00:00'` is not distinguishable from `date '1970-01-30'`

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java (hunk 2 above)
Comment (findepi) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java:212]: All the Hive View handling stuff is currently "best effort" and still very "edge case free" at this point.
Let's take care of testing edge cases as a follow up. 
We should also test with double NaN too, besides "historical dates".

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java [hunk 5]
```diff
@@ -172,4 +172,96 @@ public void testLateralViewJsonTupleAs()
...
+                "  CAST(2592000.123 AS decimal(10,3)) source_decimal_three," +
+                "  CAST(2592000 AS DECIMAL(10,0)) source_decimal_zero," +
+                "  timestamp '1970-01-30 16:00:00' source_timestamp, " +
+                "  date '1970-01-30' source_date ");
+
+        onHive().executeQuery("DROP VIEW IF EXISTS test_from_utc_timestamp_view");
+        onHive().executeQuery("CREATE VIEW " +
+                "test_from_utc_timestamp_view " +
+                "AS SELECT " +
+                // TODO: TZ identifiers like 'PST' is not supported by Trino; possibly we need a translation map (???)
```
Comment (hashhar) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java:196]: Update comment to maybe point to the new function added? And maybe a GitHub issue?

**Review by hashhar — APPROVED** [review]
Looks good % nits.

**Pull request merged**